pandas/tests/copy_view/test_methods.py

```diff
@@ -748,6 +748,57 @@ def test_head_tail(method, using_copy_on_write):
     tm.assert_frame_equal(df, df_orig)
 
 
+def test_infer_objects(using_copy_on_write):
+    df = DataFrame({"a": [1, 2], "b": "c", "c": 1, "d": "x"})
```

Comment:
It might be worth adding a test where one of the object dtype columns actually gets converted? (eg change d to `"d": np.array([0, 1], dtype=object)`) To ensure this column owns its memory (and cover that part of the code additions)

Reply:
Good point, adjusted d in the two tests below to cover both cases. 

Good to go apart from that? ``convert`` needs cow for a couple of other things